app/models/drink.rb

```diff
@@ -1,8 +1,18 @@
 class Drink < ApplicationRecord
-  validates_presence_of :name
+  validates_presence_of :name, :steps
 
   belongs_to :bar
   has_many :ingredients, dependent: :destroy
   
   accepts_nested_attributes_for :ingredients, allow_destroy: true
+
+  validate :has_ingredients
+
+  private
+
+  def has_ingredients
```

Comment:
Would a [validate associated](https://guides.rubyonrails.org/v5.2/active_record_validations.html#validates-associated) or a [validates presence for the relationship](https://guides.rubyonrails.org/v5.2/active_record_validations.html#presence) work here rather than building a custom? If not, the customer looks good and is accomplishing the right thing.